From 11e67af5214d6cc1e93892222929a3e874cc4c38 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Mon, 20 Feb 2023 00:47:50 +0100
Subject: [PATCH 73/73] arm64: dts: imx8mp: Adjust PDK3 DT to match downstream
 DTSI

The downstream kernel fork DTSI contain various differently named nodes
and custom properties, adjust the mainline DT to match the downstream
DTSI.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 .../imx8mp-dhcom-pdk3-downstream.dtsi         | 19 +++++++++++++++++++
 ...ay-nxp-spf-29678-c1-ov5640-downstream.dtsi | 18 ++++++++++++++++++
 ...k3-overlay-nxp-spf-29678-c1-ov5640-x29.dts | 19 ++++++++++---------
 ...k3-overlay-nxp-spf-29678-c1-ov5640-x31.dts | 19 ++++++++++---------
 ...-pdk3-overlay-nxp-spf-29678-c1-ov5640.dtsi |  2 ++
 .../boot/dts/freescale/imx8mp-dhcom-pdk3.dts  |  2 +-
 6 files changed, 60 insertions(+), 19 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-downstream.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-downstream.dtsi

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-downstream.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-downstream.dtsi
new file mode 100644
index 0000000000000..0473dd8ad543e
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-downstream.dtsi
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+
+#include "imx8mp-dhcom-pdkx-downstream.dtsi"
+
+&pcie_phy {
+	ext_osc = <1>;
+};
+
+&pcie {
+	clocks = <&clk IMX8MP_CLK_HSIO_ROOT>,
+		 <&clk IMX8MP_CLK_PCIE_AUX>,
+		 <&clk IMX8MP_CLK_HSIO_AXI>,
+		 <&clk_pcie>;
+	clock-names = "pcie", "pcie_aux", "pcie_phy", "pcie_bus";
+	ext_osc = <1>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-downstream.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-downstream.dtsi
new file mode 100644
index 0000000000000..5ebdccdebfb7d
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-downstream.dtsi
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+
+&cameradev {
+	status = "okay";
+};
+
+&isi_0 {
+	cap_device {
+		status = "okay";
+	};
+
+	m2m_device {
+		status = "okay";
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x29.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x29.dts
index 0907c42e09018..a6f22f5d513cd 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x29.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x29.dts
@@ -10,16 +10,17 @@
 &mipi_csi_1 {
 	status = "okay";
 
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
 
-		port@0 {
-			mipi_csi0_ep: endpoint {
-				remote-endpoint = <&ov5640_to_mipi_csi2>;
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-			};
+	port@0 {
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&ov5640_to_mipi_csi2>;
+			clock-lanes = <0>;
+			data-lanes = <2>;
+			csis-hs-settle = <13>;
+			csis-clk-settle = <2>;
+			csis-wclk;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x31.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x31.dts
index f4609131768b3..2cae0338ccdf1 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x31.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-x31.dts
@@ -10,16 +10,17 @@
 &mipi_csi_0 {
 	status = "okay";
 
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
 
-		port@0 {
-			mipi_csi0_ep: endpoint {
-				remote-endpoint = <&ov5640_to_mipi_csi2>;
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-			};
+	port@0 {
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&ov5640_to_mipi_csi2>;
+			clock-lanes = <0>;
+			data-lanes = <2>;
+			csis-hs-settle = <13>;
+			csis-clk-settle = <2>;
+			csis-wclk;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640.dtsi
index 5d73373f58caf..2b8a5ab87da27 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640.dtsi
@@ -35,3 +35,5 @@ ov5640_to_mipi_csi2: endpoint {
 &isi_0 {
 	status = "okay";
 };
+
+#include "imx8mp-dhcom-pdk3-overlay-nxp-spf-29678-c1-ov5640-downstream.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
index 75937c5766210..9d834deea19a9 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
@@ -11,8 +11,8 @@
 /dts-v1/;
 
 #include <dt-bindings/leds/common.h>
-#include <dt-bindings/phy/phy-imx8-pcie.h>
 #include "imx8mp-dhcom-som.dtsi"
+#include "imx8mp-dhcom-pdk3-downstream.dtsi"
 
 / {
 	model = "DH electronics i.MX8M Plus DHCOM Premium Developer Kit (3)";
-- 
2.39.1

